homeassistant/components/number/__init__.py

```diff
@@ -93,8 +119,60 @@ class NumberEntityDescription(EntityDescription):
 
     max_value: float | None = None
     min_value: float | None = None
+    native_max_value: float | None = None
+    native_min_value: float | None = None
+    native_unit_of_measurement: str | None = None
+    native_step: float | None = None
     step: float | None = None
 
+    def __post_init__(self) -> None:
+        """Post initialisation processing."""
+        if (
+            self.max_value is not None
+            or self.min_value is not None
+            or self.step is not None
+            or self.unit_of_measurement is not None
+        ):
+            caller = inspect.stack()[2]
+            module = inspect.getmodule(caller[0])
+            if module and module.__file__ and "custom_components" in module.__file__:
+                report_issue = "report it to the custom component author."
+            else:
+                report_issue = (
+                    "create a bug report at "
+                    "https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue"
+                )
+            _LOGGER.warning(
+                "%s is setting deprecated attributes on an instance of "
+                "NumberEntityDescription, this is not valid and will be unsupported "
+                "from Home Assistant 2022.10. Please %s",
+                module.__name__ if module else self.__class__.__name__,
+                report_issue,
+            )
+            self.native_unit_of_measurement = self.unit_of_measurement
+
+
+def ceil_decimal(value: float, precision: float = 0) -> float:
+    """Return the ceiling of f with d decimals.
+
+    This is a simple implementation which ignores floating point inexactness.
+    """
+    if precision == 0:
+        return ceil(value)
+    factor = 10**precision
+    return ceil(value * factor) / factor
+
+
+def floor_decimal(value: float, precision: float = 0) -> float:
+    """Return the floor of f with d decimals.
+
+    This is a simple implementation which ignores floating point inexactness.
+    """
+    if precision == 0:
+        return floor(value)
```

Comment:
This will return an integer.

Reply:
Well spotted @MartinHjelmare, fixed now.